✍️商品購入確認ページの追加

98hira opened the pull request:
## 概要
下図に示す商品購入確認ページのマークアップを実装する。
![スクリーンショット 2019-03-20 13 27 40](https://user-images.githubusercontent.com/32605128/54659228-46c11980-4b14-11e9-9f2f-d045c7678f7b.png)

## ToDo
- [x] 本家メルカリサイトの分析
- [x] 簡単な構成図を作成
- [x] コーディング

98hira committed:
first commit

98hira committed:
✍商品購入確認ページの追加

[time]2019/03/22-07:33️

98hira committed:
😓商品購入確認ページ以外に適用されてしまう、CSS設定を行なっていたので修正した

[問題点]
商品購入確認ページ(itemconfirm)用のCSS設定が、クレジットカード登録画面(creditregistration)にまで適用されいていた。

[原因]
現状の実装ではビュー毎に適用するCSSファイルは分けているが、ブラウザでは１つの大きなCSSファイルとして読み込まれる。
よって、ビューに適用するCSSファイルには必ず、ビュー毎に決められたclass名で指定すべきところを、bodyタグ指定でCSSを当てていた。
これによって、商品購入確認ページ(itemconfirm)の次に読み込まれるクレジットカード登録画面(creditregistration)にまで適用さていたのが原因。

[対策]
bodyで指定していた背景部分をdivタグで指定するようにした。

[time]2019/03/22-08:26

98hira commented:
実装したビューの確認結果です。
![スクリーンショット 2019-03-22 08 31 24](https://user-images.githubusercontent.com/32605128/54791364-e8fc1100-4c7c-11e9-8510-c99f17fd9eb0.png)

98hira commented on app/assets/stylesheets/modules/itemconfirm.scss:
```diff
@@ -1,41 +1,41 @@
-      font-size:24px;
+.ib-content {
+  @mixin ib_change-link() {
+    font-size: 14px;
```
**認識の確認**

上記のように記述すると、`ib_change-link()` は `.ib-content`のスコープ内でしか呼び出せない。
という認識なんですが、あっているでしょうか

miyagit replied:
うーーん、mixinのそーいう記述のしかたは正直言って初めて見るので、分からないですね。。
そもそもここのcssはうまく反映されていますか？？

僕がmixinを使うときは_mixin.scssというファイルを作って

@mixin ib_change-link {
  font-size: 14px;
  text-align: right;
}

と定義し、

上のcssを定義したいところに

@include ellipsis;と書きます。今回でいうと

.ib-content {
  @include ellipsis;

と書きますね。

98hira replied:
> そもそもここのcssはうまく反映されていますか？？

念のため再確認してみたのですが、反映されてました。

現状の開発方針が「ビュー毎に適用するCSSファイルは分ける」だったので、mixinもCSSファイル内で完結した方が良いと考えてこのような実装にしたのですが。。

ご指摘いただいた内容に関して、次のように解釈したのですが認識あっているでしょうか？

mixinを使う本質は共通化できる設定を１箇所に集めてソースの可読性を上げること。よって、今回のような書き方は一般的でない。

miyagit replied:
そうですね！一般的ではないと思います。
cssのメソッドみたいなふうに考えてもられればなと思います！

98hira commented:
ビューを作成したので、レビューをお願い致します。

一点認識あっているか確認したい項目がありますので、合わせて確認お願い致します。

98hira commented:
ビューを作成したので、レビューをお願い致します。

合わせて認識があっているか確認したい項目を https://github.com/YKKISKW/freemarket_sample_46c/pull/32#discussion_r267995875 に記載しましたので、合わせて確認お願い致します。

98hira commented:
https://github.com/YKKISKW/freemarket_sample_46c/pull/32#discussion_r268026207 に記載しましたので、ご確認お願い致します。

98hira committed:
😓商品購入確認ページ用のmixinの書き方が間違っていたので修正した

mixinの定義は/mixin直下にファイルを作成して管理するようにした。
追加したmixin定義ファイル(_itemconfirm.scss)の名前は現状ではitemconfirm.scssしか使わないため、仮で付けている。

[time]2019/03/22-16:40

98hira commented:
ご指摘頂いたmixinの書き方について修正したので確認お願いします。
701bfbc

40213151 commented:
LGTMです！

98hira committed:
コンフリクト対応

98hira committed:
Merge branch 'master' into add_itemconfirm-view
